sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -1248,7 +1261,13 @@ private void channelCleanUp(AsynchronousFileChannel channel) {
      * @param data The data which will upload to the storage file.
      * @param length Specifies the number of bytes being transmitted in the request body.
      * @return A response that only contains headers and response status code
+     *
+     * @deprecated Deprecated in favor of
+     * {@link ShareFileAsyncClient#uploadRange(ShareFileUploadRangeOptions)} and
+     * {@link ShareFileAsyncClient#uploadBufferedRange(ShareFileUploadBufferedRangeOptions)}.
+     * The former emulates this method while the latter buffers and uploads in parallel.
      */
+    @Deprecated
```

Comment:
I'm not sure we should deprecate this. Should we have some api that's simple for uploads? (in case someone doesn't care about range?

Reply:
uploadRange is basically this simple upload API

Reply:
We dont want people to be confused thinking this API also supports any size data (like we do for our other libraries)

Reply:
I mean. I have to do `uploadRange(new XOptions(data, length))` . I was hoping to have `uploadRange(data, lenght)` (which is what old upload is). But now that's old upload is just delegating to new api - should it just stay as is ?

Reply:
I deprecated based on the changes you made in .NET for DownloadContent and DownloadStreaming. The thing is that the upload method is already an upload range method, and the buffered method also needs to be capable of ranges for people who want to upload ranges larger than 4MB. We struggled with a way to name a new method while keeping the old method.

We could have the simpler arguments for the min overload like Gauri suggests in another comment, where just the data stream is needed (and length for sync clients until a better InputStream->Flux converted exists).

Reply:
Strategy resolved elsewhere.